frontends/concrete-python/examples/xor_distance/hamming_distance_to_clear.py

```diff
@@ -0,0 +1,163 @@
+import argparse
+import time
+
+import numpy as np
+
+from concrete import fhe
+
+
+# Hamming weight computation
+def hw(x):
+    # Hamming Weight table for 8b entries
+    hw_table_ref = [np.binary_repr(x).count("1") for x in range(2**8)]
+
+    # fmt: off
+    assert np.array_equal(hw_table_ref, [
+        0, 1, 1, 2, 1, 2, 2, 3, 1, 2, 2, 3, 2, 3, 3, 4, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3,
+        4, 3, 4, 4, 5, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4,
+        4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 1, 2, 2, 3, 2, 3, 3, 4, 2, 3, 3, 4, 3, 4, 4, 5, 2,
+        3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5,
+        4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 1, 2, 2, 3, 2, 3, 3,
+        4, 2, 3, 3, 4, 3, 4, 4, 5, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 2, 3,
+        3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5,
+        6, 6, 7, 2, 3, 3, 4, 3, 4, 4, 5, 3, 4, 4, 5, 4, 5, 5, 6, 3, 4, 4, 5, 4, 5, 5, 6,
+        4, 5, 5, 6, 5, 6, 6, 7, 3, 4, 4, 5, 4, 5, 5, 6, 4, 5, 5, 6, 5, 6, 6, 7, 4, 5, 5,
+        6, 5, 6, 6, 7, 5, 6, 6, 7, 6, 7, 7, 8])
+    # fmt: on
+
+    hw_table = fhe.LookupTable(hw_table_ref)
+    return hw_table[x]
+
+
+def mapme(x):
+    # Map 0 to -1, and keep 1 as 1
+    return 2 * x - 1
+
+
+# Reference function for tests
+def dist_in_clear(x, y):
+    return np.sum(hw(x ^ y))
+
+
+# In FHE
+def dist_in_fhe(x_mapped, y_mapped):
+
+    # x is a line tensor, whose 0's have been replaced by -1
+    # y_clear is a column tensor, whose 0's have been replaced by -1
+    assert x_mapped.ndim == y_mapped.ndim == 2
+    assert x_mapped.shape[0] == y_mapped.shape[1] == 1
+
+    u = np.matmul(x_mapped, y_mapped)[0][0]
+
+    # So, u is a scalar:
+    # - bits which are the same between x and y_clear (either two -1's or two 1's) count for a +1 in the scalar
+    # - bits which are different between x and y_clear (either (-1, 1) or (1, -1)) count for a -1 in the scalar
+    # Hence the HW distance is (len(x) - u) / 2
+    final_result = np.prod(x_mapped.shape) - u
+
+    # The result which is returned is the double of the distance, we'll halve this in the clear
+    return final_result
+
+
+# Manage user args
+def manage_args():
```

Comment:
What about using the documentation style we have in the repo?
```python
def manage_args() -> argparse.ArgumentParser:
    """
    Manage user args.

    Returns:
        argparse.ArgumentParser:
            parser for the arguments of the script
    """
```

Reply:
Yes sure, but again, it's in a part which was already reviewed

Reply:
Done